core/src/main/java/com/graphhopper/routing/util/DataFlagEncoder.java

```diff
@@ -327,6 +357,96 @@ public long handleWayTags(ReaderWay way, long allowed, long relationFlags) {
         }
     }
 
+    private long extractMeter(ReaderWay way, long flags, EncodedDoubleValue valueEncoder, List<String> keys) {
+        String value = extractRoadAttribute(way, keys);
+
+        if (Helper.isEmpty(value)) return flags;
+
+        double val;
+        try {
+            val = stringToMeter(value);
+        } catch (Throwable t) {
+            LOG.warn("Unable to extract meter from malformed road attribute '{}' for way (OSM_ID = {}).", value, way.getId(), t);
+            val = 0.0;
+        }
+
+        return valueEncoder.setDoubleValue(flags, val);
+    }
+
+    private long extractTons(ReaderWay way, long flags, EncodedDoubleValue valueEncoder, List<String> keys) {
+        String value = extractRoadAttribute(way, keys);
+
+        if (Helper.isEmpty(value)) return flags;
+
+        double val;
+        try {
+            val = stringToTons(value);
+        } catch (Throwable t) {
+            LOG.warn("Unable to extract tons from malformed road attribute '{}' for way (OSM_ID = {}).", value, way.getId(), t);
+            val = 0.0;
+        }
+
+        return valueEncoder.setDoubleValue(flags, val);
+    }
+
+    private String extractRoadAttribute(ReaderWay way, List<String> keys) {
+        String value = null;
+
+        for (String key : keys) {
+            value = way.getTag(key);
+            if (!Helper.isEmpty(value)) break;
+        }
+
+        return value;
+    }
+
+    double stringToTons(String value) {
+        value = value.toLowerCase().replaceAll(" ", "").replaceAll("(tons|ton)", "t");
+        double factor = 1;
+        if (value.endsWith("t")) {
+            value = value.substring(0, value.length() - 1);
+        } else if (value.endsWith("lbs")) {
+            value = value.substring(0, value.length() - 3);
+            factor = 0.00045359237;
+        }
+
+        return Double.parseDouble(value) * factor;
+    }
+
+    double stringToMeter(String value) {
+        value = value.toLowerCase().replaceAll(" ", "").replaceAll("(meters|meter|mtrs|mtr|mt|m\\.)", "m");
```

Comment:
According to the [Wiki](http://wiki.openstreetmap.org/wiki/Map_Features/Units) these vales might also happen: `	metre, metres, meter, meters`. Couldn't find them via Taginfo, so not sure if this is relevant.